### PR TITLE
Cygwin build fixes.

### PR DESCRIPTION
**Pull request by tsulej**
Fixes all issues reported here: https://github.com/mmp/pbrt-v3/issues/155

What is done here:
1. Minimum CMake bumped to 2.8.4 to remove long warning text under Cygwin
2. `noinline` parentheses escaped
3. Changed order of local thread keyword to select `__thread` first. `__declspec(thread)` is ignored under Cygwin.

### CODE REVIEW

**Comment by tsulej** [comment]
It seems that escaping parentheses breaks build under Visual Studio.

File: CMakeLists.txt [hunk 1]
```diff
@@ -186,7 +186,7 @@ int main() { }"
 HAVE_ATTRIBUTE_NOINLINE )
 
 IF ( HAVE_DECLSPEC_NOINLINE )
-  ADD_DEFINITIONS ( -D "PBRT_NOINLINE=__declspec(noinline)" )
+  ADD_DEFINITIONS ( -D "PBRT_NOINLINE=__declspec\\(noinline\\)" )
```
Comment (mmp) [inline CMakeLists.txt]: Unfortunately this breaks the MSVC build. :-(. I'm curious why it matters for Cygwin. Would it work to check HAVE_ATTRIBUTE_NOINLINE first, under the theory that should also be true for cygwin?

Comment (tsulej) [reply CMakeLists.txt]: The problem is with `sh` which treats parentheses as piece of it's script. Checking if reorder helps.

**Comment by mmp** [comment]
Thanks!

**Pull request merged**